plugins/fastem_roa_import_export.py

```diff
@@ -0,0 +1,341 @@
+        :param app: A reference to the class App object.
+        :param project_list_controller: (FastEMProjectListController) class which creates/removes new
+            FastEM projects.
+
+        """
+        if not self.initialized:
+            self.init()
+
+        parent = self._app.GetTopWindow()
+        if not self._frame:
+            self._frame = ImportExportFrame(
+                parent=parent,
+                pos=self._pos,
+                size=self._size,
+                project_list_controller=project_list_controller,
+            )
+
+        self._frame.Show()
+        if self._frame.IsIconized():
+            self._frame.Iconize(False)
+        self._frame.Raise()
+
+
+class ImportExportFrame(wx.Frame):
+    """
+    This class is the frame that holds the Import Export ROAs tool. The toolbar is managed here.
+
+    :param project_list_controller: (FastEMProjectListController) class which creates/removes new
+            FastEM projects.
+
+    """
+
+    def __init__(self, title="Import Export ROAs Tool", project_list_controller=None, *args, **kw):
+        kw["title"] = title
+        wx.Frame.__init__(self, *args, **kw)
+
+        self._project_list_controller = project_list_controller
+
+        self.make_tool_bar()
+
+        self.Bind(wx.EVT_CLOSE, self.on_close)
+        if self.Parent:
+            tlw = self.Parent.GetTopLevelParent()
+            tlw.Bind(wx.EVT_CLOSE, self.on_close)
+
+    def make_tool_bar(self):
+        """Create the Import Export ROAs toolbar."""
+        tbar = self.CreateToolBar(wx.TB_HORIZONTAL | wx.TB_FLAT | wx.TB_TEXT | wx.NO_BORDER)
+        tbar.SetToolBitmapSize(wx.Size(24, 24))
+
+        import_bmp = IMPORT_IMAGE.GetBitmap()
+        export_bmp = EXPORT_IMAGE.GetBitmap()
+
+        import_tool = tbar.AddTool(-1, "Import", import_bmp, shortHelp="Import ROAs")
+        export_tool = tbar.AddTool(-1, "Export", export_bmp, shortHelp="Export ROAs")
+        tbar.Realize()
+
+        self.Bind(wx.EVT_TOOL, self.on_import, import_tool)
+        self.Bind(wx.EVT_TOOL, self.on_export, export_tool)
+
+    def on_import(self, evt):
+        """On Import ROA tool click."""
+        evt.Skip()
+        with wx.FileDialog(
+            self,
+            "Import ROAs from a CSV file",
+            wildcard="CSV files (*.csv)|*.csv",
+            style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST,
+        ) as fileDialog:
+            if fileDialog.ShowModal() == wx.ID_CANCEL:
+                return  # the user changed their mind
+
+            # Proceed loading the file chosen by the user
+            pathname = fileDialog.GetPath()
+        try:
+            with open(pathname, newline="") as csvfile:
+                reader = csv.DictReader(csvfile)
+                data = {}
+                # Firstly extract the csv file data in a structured way
+                # {"project_name": [roa]} where roa is dict containing
+                # roa, roc_2, roc_3 name and its coordinates respectively
```

Comment:
Why using CSV format? It's a very limited format, so you end up having to use complex code to parse it (special Enum). I'd suggest to use the same format that we used to save the "CryoFeatures`: JSON. See odemis.acq.features. It's much more adapted to store the sort of structure which is used here.

Using JSON should also be much more future-proof in case the data changes. It'll be easier to detect the file uses an older format.

Reply:
The idea of using csv is that the user can easily copy and paste ROAs and adjust the position. A possible workflow is that the user draws 3 ROAs on a ribbon, then exports the data to csv, extrapolates the positions of these 3 ROAs to define the rest of the ribbon, imports it again and only has to move the new ROAs a bit.

Reply:
I see. If the choice of CSV was deliberate, and works better for the user, let's go for it!